test/remote-bitfield.js

```diff
@@ -58,6 +58,70 @@ test('remote congituous length consistency (remote-bitfield findFirst edge case)
   )
 })
 
+test.solo('bitfield messages sent on cache miss', async function (t) {
+  const original = await create(t)
+  const sparse = await create(t, original.key)
+  const empty = await create(t, original.key)
+
+  await original.append(['a', 'b', 'c', 'd', 'e'])
+
+  replicate(original, sparse, t)
+  await original.get(2)
+  await original.get(3)
+
+  replicate(sparse, empty, t)
+  await new Promise(resolve => setTimeout(resolve, 1000))
+
+  t.is(empty.replicator.peers.length, 1, 'Sanity check')
+  const stats = empty.replicator.peers[0].stats
+  t.is(stats.wireBitfield.rx, 0, 'initially no bitfields sent (sanity check')
+
+  await t.exception(
+    async () => {
+      await empty.get(1, { timeout: 100 })
+    },
+    /REQUEST_TIMEOUT/,
+    'request on unavailable block times out (sanity check)'
+  )
+  t.is(stats.wireBitfield.rx, 1, 'Requests bitfield on cache miss')
+  console.log(stats)
+
+  await t.exception(
+    async () => {
+      await empty.get(0, { timeout: 100 })
+    },
+    /REQUEST_TIMEOUT/,
+    'request on unavailable block times out (sanity check)'
+  )
+  t.is(stats.wireBitfield.rx, 1, 'Bitfield not sent again if there were no changes')
+
+  await t.exception(
+    async () => {
+      await empty.get(0, { timeout: 100 })
+    },
+    /REQUEST_TIMEOUT/,
+    'request on unavailable block times out (sanity check)'
+  )
+
+  await original.append(['f', 'g', 'h'])
+  await sparse.get(6)
+  await new Promise(resolve => setTimeout(resolve, 1000))
+
+  t.is(stats.wireBitfield.rx, 1, 'New bitfield not aggressively sent')
+
+  await t.exception(
+    async () => {
+      await empty.get(5, { timeout: 100 })
+    },
+    /REQUEST_TIMEOUT/,
+    'request on unavailable block times out (sanity check)'
+  )
+
+  // TODO: verify this is expected
+  t.is(stats.wireBitfield.rx, 1, 'No new bitfield message sent on new cache miss')
```

Comment:
Is this expected? I had expected a new bitfield message, where the sparse peer communicates which blocks it now has available (both peers did send a new wireSync message to each other, so they're aware of each other's lengths)

Reply:
confirmed it's expected, so just got rid of all those additional asserts